### ISSUE TITLE
SFad requires a lot more memory than SLFads of the same size

### ISSUE THREAD

**Issue by bathmatt** [trilinos/Trilinos#450]
@rppawlo @eric-c-cyr @etphipp 
This is reporting the issue for the SFad vs the SLFad and memory requirements.

Here is the issue summary, when I set 
-D Panzer_ENABLE_FADTYPE:STRING="Sacado::Fad::SLFad<double,50>" \

I can use twice the sized workset than if I set the FAD type to
-D Panzer_ENABLE_FADTYPE:STRING="Sacado::Fad::SFad<double,50>" \

This is on Cuda ,haven't tested on CPU.  I suspect there is some duplication of memory but not sure.  I ran the whole code through intel inspector and it didn't find any memory issues of note.

**Comment by bathmatt** [comment]
I confirmed this with gcc, with SLFad, 3.8G, with SFad 6.3G.  I don't see how the bug can be in panzer honestly, but wouldn't rule it out completely..

**Comment by etphipp** [comment]
Can you tell me (offline if necessary) in which code you are seeing this?  A test in Panzer or an application code?  This has got to be a bug, possibly in the view specialization for Fad.

**Comment by rppawlo** [comment]
FYI, panzer allows you to specify the fad type at configure time:
```
TRIBITS_ADD_OPTION_AND_DEFINE(${PACKAGE_NAME}_ENABLE_FADTYPE
  PANZER_FADTYPE
  "Choose the Sacado automatic differentiation scalart type (default is DFad)."
  "Sacado::Fad::DFad<RealType>" )
```

This is then used in the disc-fe/src/Panzer_Traits.hpp for declaring the Jacobian and Tangent type (Hessians are handled separately and hard coded).

```
    typedef PANZER_FADTYPE FadType;
    ...
    struct Residual { typedef RealType ScalarT; };
    struct Jacobian { typedef FadType ScalarT;  };
    struct Tangent { typedef FadType ScalarT;  };
    ...
    typedef Sacado::mpl::vector< Residual
                               , Jacobian
                               , Tangent
#ifdef Panzer_BUILD_HESSIAN_SUPPORT
                               , Hessian
#endif
                                > EvalTypes;

```

I suspect any test will show this. You could try any of the tests in Panzer/adapters-stk/example/main_driver

**Comment by etphipp** [comment]
I've now looked into this a little bit, and so far I haven't been able to reproduce this.  

First I made a small test program that creates a very large view with SFad and SLFad scalar types.  I found no difference in the memory usage when running on both OpenMP and CUDA.  This suggests to me there isn't an issue with the View specialization.

I've also built Drekar with both SFad and SLFad on OpenMP, and I'm also not seeing any difference in memory usage.  I guess I'll have to build with CUDA next.
